README.md

```diff
@@ -340,190 +397,446 @@ Convert String / DataSource.Text / DataSource.JSON to the following MySQL types:
+The MySQL connector supports _model discovery_ that enables you to create
+LoopBack models based on an existing database schema. Once you defined your
+datasource:
+
+- LoopBack 4 users could use the commend
+  [`lb4 discover`](https://loopback.io/doc/en/lb4/Discovering-models.html) to
+  discover models.
+- For LB3 users, please check
+  [Discovering models from relational databases](https://loopback.io/doc/en/lb3/Discovering-models-from-relational-databases.html).
+  (See
+  [database discovery API](http://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-discoverandbuildmodels)
+  for related APIs information)
 
 ### Auto-migration
 
-The MySQL connector also supports _auto-migration_ that enables you to create a database schema
-from LoopBack models using the [LoopBack automigrate method](http://apidocs.strongloop.com/loopback-datasource-juggler/#datasource-prototype-automigrate).
+The MySQL connector also supports _auto-migration_ that enables you to create a
+database schema from LoopBack models. For example, based on the following model,
+the auto-migration method would create/alter existing `Customer` table in the
+database. Table `Customer` would have two columns: `name` and `id`, where `id`
+is also the primary key that has `auto_increment` set as it has definition of
+`type: 'Number'` and `generated: true`:
+
+```ts
+@model()
+export class Customer extends Entity {
+  @property({
+    id: true,
+    type: 'Number',
+    generated: true,
+  })
+  id: number;
+
+  @property({
+    type: 'string',
+  })
+  name: string;
+}
+```
+
+Moreover, additional MySQL-specific properties mentioned in the
+[Data mapping properties](#data-mapping-properties) section work with
+auto-migration as well.
+
+#### auto-generated ids
+
+For now LoopBack MySQL connector only supports auto-generated id
+(`generated: true`) for integer type as for MySQL, the default id type is
+_integer_. If you'd like to use other types such as string (uuid) as the id
+type, you can:
+
+- use uuid that is **generated by your LB application** by setting
+  [`defaultFn: uuid`](https://loopback.io/doc/en/lb4/Model.html#property-decorator).
+
+```ts
+  @property({
+    id: true,
+    type: 'string'
+    defaultFn: 'uuidv4',
+    // generated: true,  -> not needed
+  })
+  id: string;
+```
 
-For more information on auto-migration, see [Creating a database schema from models](https://loopback.io/doc/en/lb3/Creating-a-database-schema-from-models.html) for more information.
+- Alter the table in your database to use a certain function if you prefer
+  having the database to generate the value.
+
+```ts
+  @property({
+    id: true,
+    type: 'string'
+    // generated: true,  -> not needed
+  })
+  id: string;
```

Comment:
if there is no `generated: true` and not making this optional, would we have the situation that the POST call expects the `id` property to exist while it should be generated?

Reply:
You're right, this should be
```
    generated: true,
    useDefaultIdType: false,
```
👍 thanks!